internal/pkg/runtime/engines/singularity/rpc/server/server_linux.go

```diff
@@ -134,6 +185,7 @@ func (t *Methods) LoopDevice(arguments *args.LoopArgs, reply *int) error {
 	} else {
 		var err error
 		image, err = os.OpenFile(arguments.Image, arguments.Mode, 0600)
+		defer image.Close()
```

Comment:
is this necessary? "image" is used below as an argument to loopdev.AttachFromFile, and that sets things up so that this file descriptor is closed. There's no explicit call to image.Close(), that's right. And there might be some error paths where this is not closed, too.

Did you find a case where this was needed?

Reply:
Didn't really find any cases that are going . wrong with this image not being closed. This change has nothing to do with the overall change. I just thought image should be closed before we exit this function.